erpnext/erpnext_integrations/doctype/tally_migration/tally_migration.py

```diff
@@ -0,0 +1,512 @@
+			for account in accounts:
+				if account["NAME"] in (self.tally_creditors_account, self.tally_debtors_account):
+					yield get_parent(account), account["NAME"], 0
+				else:
+					yield get_parent(account), account["NAME"], 1
+
+		def get_ledgers(accounts):
+			for account in accounts:
+				# If Ledger doesn't have PARENT field then don't create Account
+				# For example "Profit & Loss A/c"
+				if account.PARENT:
+					yield account.PARENT.string, account["NAME"], 0
+
+		def get_parent(account):
+			if account.PARENT:
+				return account.PARENT.string
+			return {
+				("Yes", "No"): "Application of Funds (Assets)",
+				("Yes", "Yes"): "Expenses",
+				("No", "Yes"): "Income",
+				("No", "No"): "Source of Funds (Liabilities)",
+			}[(account.ISDEEMEDPOSITIVE.string, account.ISREVENUE.string)]
+
+		def get_children_and_parent_dict(accounts):
+			children, parents = {}, {}
+			for parent, account, is_group in accounts:
+				children.setdefault(parent, set()).add(account)
+				parents.setdefault(account, set()).add(parent)
+				parents[account].update(parents.get(parent, []))
+			return children, parents
+
+		def remove_parties(parents, children, group_set):
+			customers, suppliers = set(), set()
+			for account in parents:
+				if self.tally_creditors_account in parents[account]:
+					children.pop(account, None)
+					if account not in group_set:
+						suppliers.add(account)
+				elif self.tally_debtors_account in parents[account]:
+					children.pop(account, None)
+					if account not in group_set:
+						customers.add(account)
+			return children, customers, suppliers
+
+		def traverse(tree, children, accounts, roots, group_set):
+			for account in accounts:
+				if account in group_set or account in roots:
+					if account in children:
+						tree[account] = traverse({}, children, children[account], roots, group_set)
+					else:
+						tree[account] = {"is_group": 1}
+				else:
+					tree[account] = {}
+			return tree
+
+		def get_parties_addresses(collection, customers, suppliers):
+			parties, addresses = [], []
+			for account in collection.find_all("LEDGER"):
+				party_type = None
+				if account.NAME.string in customers:
+					party_type = "Customer"
+					parties.append({
+						"doctype": party_type,
+						"customer_name": account.NAME.string,
+						"tax_id": account.INCOMETAXNUMBER.string if account.INCOMETAXNUMBER else None,
+						"customer_group": "All Customer Groups",
+						"territory": "All Territories",
+						"customer_type": "Individual",
+					})
+				elif account.NAME.string in suppliers:
+					party_type = "Supplier"
+					parties.append({
+						"doctype": party_type,
+						"supplier_name": account.NAME.string,
+						"pan": account.INCOMETAXNUMBER.string if account.INCOMETAXNUMBER else None,
+						"supplier_group": "All Supplier Groups",
+						"supplier_type": "Individual",
+					})
+				if party_type:
+					address = "\n".join([a.string for a in account.find_all("ADDRESS")[:2]])
```

Comment:
Taking only the first two lines of address can lead to losing information... tally allows unlimited lines for the address. can you please take the whole address and put it into `address_line1` and `address_line2`.